Assets/Scripts/Goal.cs

```diff
@@ -32,6 +34,7 @@ enum GoalState {
             playerNumber = Utils.PlayerNumber.TWO;
         }
         ResetGoal();
+        InitGoalIndicator();
```

Comment:
Since the goal indicator is only for self to see (it looks weird from the opponent's viewpoint), perhaps you can wrap this with something like
```
PhotonView pv = GetComponent<PhotonView>();
if (!PhotonNetwork.IsConnected || pv.IsMine) {
    InitGoalIndicator();
}
```
and do the same check for `HandleGoalIndicator();` in `Update`